router-tests/header_propagation_test.go

```diff
@@ -0,0 +1,521 @@
+			})
+		})
+
+		// Test case for Expires header: earliest expiration wins
+		t.Run("global earliest Expires wins", func(t *testing.T) {
+			t.Parallel()
+			testenv.Run(t, &testenv.Config{
+				RouterOptions: global(config.ResponseHeaderRuleAlgorithmMostRestrictiveCacheControl, ""),
+				Subgraphs:     subgraphsWithExpiresHeader,
+			}, func(t *testing.T, xEnv *testenv.Environment) {
+				res := xEnv.MakeGraphQLRequestOK(testenv.GraphQLRequest{
+					Query: queryEmployeeWithHobby,
+				})
+				expires := res.Response.Header.Get("Expires")
+				require.NotEmpty(t, expires)
+
+				// Parse the Expires header and convert both times to UTC for comparison
+				parsedExpires, err := http.ParseTime(expires)
+				require.NoError(t, err)
+
+				now := time.Now().Add(5 * time.Minute)                        // Example expiration
+				require.WithinDuration(t, now, parsedExpires, 20*time.Second) // Ensure expiration is within expected range
+				require.Equal(t, `{"data":{"employee":{"id":1,"hobbies":[{},{"name":"Counter Strike"},{},{},{}]}}}`, res.Body)
+			})
+		})
+
+		t.Run("default value adds max age", func(t *testing.T) {
+			t.Parallel()
+
+			t.Run("global default age sets for all requests", func(t *testing.T) {
+				t.Parallel()
+				testenv.Run(t, &testenv.Config{
+					RouterOptions: []core.Option{
+						core.WithHeaderRules(config.HeaderRules{
+							All: &config.GlobalHeaderRule{
+								Response: []*config.ResponseHeaderRule{
+									{
+										Operation: config.HeaderRuleOperationPropagate,
+										Algorithm: config.ResponseHeaderRuleAlgorithmMostRestrictiveCacheControl,
+										Default:   "max-age=300",
+									},
+								},
+							},
+						})},
+					Subgraphs: cacheOptions("", ""),
+				}, func(t *testing.T, xEnv *testenv.Environment) {
+					res := xEnv.MakeGraphQLRequestOK(testenv.GraphQLRequest{
+						Query: queryEmployeeWithHobby,
+					})
+					cc := res.Response.Header.Get("Cache-Control")
+					require.Equal(t, "max-age=300", cc) // Shorter max-age wins
+					require.Equal(t, `{"data":{"employee":{"id":1,"hobbies":[{},{"name":"Counter Strike"},{},{},{}]}}}`, res.Body)
+				})
+			})
+
+			t.Run("partial default age sets for requests with information", func(t *testing.T) {
+				t.Parallel()
+				alg := config.ResponseHeaderRuleAlgorithmMostRestrictiveCacheControl
+				testenv.Run(t, &testenv.Config{
+					RouterOptions: []core.Option{
+						core.WithHeaderRules(config.HeaderRules{
+							Subgraphs: map[string]*config.GlobalHeaderRule{
+								"employees": {
+									Response: []*config.ResponseHeaderRule{
+										getRule(alg, ""),
+									},
+								},
+								"hobbies": {
+									Response: []*config.ResponseHeaderRule{
+										{
+											Operation: config.HeaderRuleOperationPropagate,
+											Algorithm: config.ResponseHeaderRuleAlgorithmMostRestrictiveCacheControl,
+											Default:   "max-age=300",
```

Comment:
Please add 3 tests:
1. default no cache, one subgraph returns no headers -> no cache
2. default no cache, both subgraphs return max age, use most restrictive max age
3. default max age 300, both subgraphs return max age 60 -> max age 60

Reply:
I added them. however, in your case #2 (no cache, while subgraphs return max age), there the cache header will be no-cache, because it's the most restrictive cache option.